deploy/legionPipeline.groovy

```diff
@@ -0,0 +1,222 @@
+                    '''
+                    step([
+                        $class : 'RobotPublisher',
+                        outputPath : 'tests/robot/',
+                        outputFileName : "*.xml",
+                        disableArchiveOutput : false,
+                        passThreshold : 100,
+                        unstableThreshold: 95.0,
+                        onlyCritical : true,
+                        otherFiles : "*.png",
+                    ])
+                }
+                junit 'tests/python/nosetests.xml'
+    }
+    else {
+        println('Skipped due to UseRegressionTests property')
+    }
+}
+
+def deployLegionEnclave() {
+    dir('deploy/ansible'){
+        withCredentials([file(credentialsId: params.Profile, variable: 'CREDENTIAL_SECRETS')]) {
+            withAWS(credentials: 'kops') {
+                wrap([$class: 'AnsiColorBuildWrapper', colorMapName: "xterm"]) {
+                    ansiblePlaybook(
+                        playbook: 'deploy-legion-enclave.yml',
+                        extras: ' --extra-vars "profile=${Profile} base_version=${BaseVersion}  local_version=${LocalVersion} enclave_name=${EnclaveName}"',
+                        colorized: true
+                    )
+                }
+            }
+        }
+    }
+}
+
+def terminateLegionEnclave() {
+    dir('deploy/ansible'){
+        withCredentials([file(credentialsId: params.Profile, variable: 'CREDENTIAL_SECRETS')]) {
+            withAWS(credentials: 'kops') {
+                wrap([$class: 'AnsiColorBuildWrapper', colorMapName: "xterm"]) {
+                    ansiblePlaybook(
+                        playbook: 'terminate-legion-enclave.yml',
+                            extras: ' --extra-vars "profile=${Profile} enclave_name=${EnclaveName}"',
+                            colorized: true
+                        )
+                }
+            }
+        }
+    }
+}
+
+def notifyBuild(String buildStatus = 'STARTED') {
+    // build status of null means successful
+    buildStatus =  buildStatus ?: 'SUCCESSFUL'
+
+    def previousBuild = currentBuild.getPreviousBuild()
+    def previousBuildResult = previousBuild != null ? previousBuild.result : null
+
+    def currentBuildResultSuccessful = buildStatus == 'SUCCESSFUL' || buildStatus == 'SUCCESS'
+    def previousBuildResultSuccessful = previousBuildResult == 'SUCCESSFUL' || previousBuildResult == 'SUCCESS'
+
+    def masterOrDevelopBuild = params.GitBranch == 'origin/develop' || params.GitBranch == 'origin/master'
+
+    print("NOW SUCCESSFUL: ${currentBuildResultSuccessful}, PREV SUCCESSFUL: ${previousBuildResultSuccessful}, MASTER OR DEV: ${masterOrDevelopBuild}")
+
+    if (!masterOrDevelopBuild)
+        return
+
+    // Skip green -> green
+    if (currentBuildResultSuccessful && previousBuildResultSuccessful)
+        return
+
+    // Default values
+    def colorCode = '#FF0000'
+    def summary = """\
```

Comment:
We should use different messages for each Jenkins job because they contain different parameters to print

Reply:
fixed